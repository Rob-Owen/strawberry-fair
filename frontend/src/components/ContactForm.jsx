import React, { useState } from 'react'
import PropTypes from 'prop-types'
import _ from 'lodash'

import contactFormConfig from '../data/contactForm.config'

const requiredErrorMessage = 'This field is required'

const SUBMISSION_STATUS = { SUBMITTED: 'SUBMITTED', UNSUBMITTED: 'UNSUBMITTED', SUBMISSION_FAILED: 'SUBMISSION_FAILED' }

const ContactForm = () => {
  const [status, setStatus] = useState(SUBMISSION_STATUS.UNSUBMITTED)

  const [name, nameError, setName] = useValidatedState("", {
    required: requiredErrorMessage,
    maxLength: {
      value: 80,
      message: 'Max length is 80 characters',
    }
  })
  
  const [email, emailError, setEmail] = useValidatedState("", {
    required: requiredErrorMessage,
    pattern: {
      value: /^\S+@\S+\.\S+$/i,
      message: 'Invalid email address'
    }
  })

  const [message, messageError, setMessage] = useValidatedState("", {required: requiredErrorMessage})

  const configFields = contactFormConfig.fields

  if (status === SUBMISSION_STATUS.SUBMITTED) {
    return (
      <h2 className="title">
        Form submitted! We&apos;ll get back to you as soon as we can.
      </h2>
    )
  }

  return (
    <form onSubmit={event => {
        setName(name)
        setEmail(email)
        setMessage(message)
        
        const data = {name, email, message}
        const errors = {nameError, emailError, messageError}
        submitForm(event, data, errors, setStatus)
      }}>
      <FormField label={configFields.name.label} error={nameError}>
        <input className="input" 
          type="text"
          placeholder="Stan Desk"
          name="name"
          onChange={event => setName(event.target.value)}
        />
      </FormField>
      <FormField label={configFields.email.label} error={emailError}>
        <input className="input"
          type="text" 
          placeholder="StanDesk@NoSitting.com" 
          name="email"
          onChange={event => setEmail(event.target.value)}
        />
        </FormField>
      <FormField label={configFields.message.label} error={messageError}>
        <textarea className="textarea"
          name="message"
          onChange={event => setMessage(event.target.value)}
        />
      </FormField>
      <button className="button has-background-primary has-text-white" type="submit">
        Submit
      </button>
      <FailedSubmissionError status={status} />
    </form>
  )
}

/**
 * Helper function for setting state with validation
 * @param {Any} initialState 
 * @param {Object} validationOptions
 * @returns {Array} [state, errorMessage, setValidatedState]
 */
const useValidatedState = (initialState, validationOptions) => {
  const [state, setState] = useState(initialState)
  const [errorMessage, setErrorMessage] = useState("")

  const setValidatedState = (newState) => {
    setState(newState)
    if (validationOptions.required && !newState) {
      setErrorMessage(_.get(validationOptions.required, "message", validationOptions.required))

    } else if (newState.length > _.get(validationOptions, 'maxLength.value')) {
      setErrorMessage(validationOptions.maxLength.message)

    } else if (_.get(validationOptions, 'pattern.value', '') instanceof RegExp && !validationOptions.pattern.value.test(newState)) {
      setErrorMessage(validationOptions.pattern.message)

    } else {
      setErrorMessage("")
    }
  }

  return [state, errorMessage, setValidatedState]
}

export default ContactForm

const submitForm = (event, data, errors, setStatus) => {
  event.preventDefault() // Prevents submission from redirecting the page

  if (
    Object.values(errors).filter(e => e !== "").length > 0 || 
    Object.values(data).filter(v => v === "").length > 0 // This is required to handle race conditions with setState
  ) {
    return console.error(errors)
  }

  let postBody = {}

  Object.keys(data).forEach(label =>  postBody[contactFormConfig.fields[label].key] = data[label])
  const queryString = toQueryString(postBody)

  try {
    fetch(contactFormConfig.url, {
      method: contactFormConfig.method,
      headers: contactFormConfig.headers,
      mode: contactFormConfig.mode,
      referrerPolicy: contactFormConfig.referrerPolicy,
      body: queryString,
    })
    .then(() => setStatus(SUBMISSION_STATUS.SUBMITTED))
    .catch(() => setStatus(SUBMISSION_STATUS.SUBMISSION_FAILED))
  } catch (error) {
    console.error(error)
    setStatus(SUBMISSION_STATUS.SUBMISSION_FAILED)
  }
}

const toQueryString = (obj) =>
  Object.keys(obj).map(key => `${key}=${encodeURIComponent(obj[key])}`).join('&')

const FormField = ({label, error, children}) => {
  const newClassName = error ? children.props.className + ' is-primary' : children.props.className
  
  return (
    <div className="field">
      <label className="label">{label}</label>
      <div className="control">
        {React.cloneElement(children, {className: newClassName})}
        <InvalidInputError error={error} />
      </div>
    </div>
  )
}

const InvalidInputError = ({error}) => {
  if (error) {
    return (
      <span className="has-text-primary">{error}</span>
    )
  } else { 
    return null
  } 
}

const FailedSubmissionError = ({status}) => {
  if (status === SUBMISSION_STATUS.SUBMISSION_FAILED) {
    return (
      <div className="has-text-primary">
        Your submission failed. Please try again or directly email us at enquiries@strawberry-fair.org.uk
      </div>
    )
  } else {
    return null
  }
}

FormField.propTypes = {
  label: PropTypes.string.isRequired,
  error: PropTypes.object,
  children: PropTypes.element.isRequired
}

InvalidInputError.propTypes = {
  error: PropTypes.object
}

FailedSubmissionError.propTypes = {
  status: PropTypes.oneOf(Object.values(SUBMISSION_STATUS))
}
