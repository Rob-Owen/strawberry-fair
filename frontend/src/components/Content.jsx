import React from 'react'
import PropTypes from 'prop-types'

export const HTMLContent = ({ content, className }) => (
    <div className={`content ${className || ''}`} dangerouslySetInnerHTML={{ __html: content }} />
)

// This component is here to make sure the HTML in a smaller frame (e.g. upcoming events) isn't too large.
// It basically renders <h1>s and <h2>s etc as <h4>s <h5>s etc.
export const HTMLContentSmall = ({ content, className }) => (
    <div className={`content is-small ${className || ''}`} dangerouslySetInnerHTML={{ __html: content }} />
)

export const Content = ({ content, className }) => (
    <div className={`content ${className || ''}`}>{content}</div>
)

Content.propTypes = {
    content: PropTypes.node,
    className: PropTypes.string,
}

HTMLContent.propTypes = Content.propTypes
HTMLContentSmall.propTypes = Content.propTypes