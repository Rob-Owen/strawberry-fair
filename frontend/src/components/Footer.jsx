import React from 'react'
import { IconContext } from 'react-icons'
import { FaFacebook, FaTwitter, FaEnvelopeOpenText } from 'react-icons/fa'
import { useStaticQuery, graphql, Link } from 'gatsby'
import footerPreviewContent from '../data/footerPreviewContent'
import { PreviewContext } from '../util/context.jsx'
import PropTypes from 'prop-types'
import HeaderButtons from './headerButtons'

export const Footer = (data) => (
    <PreviewContext.Consumer>
      {value => <FooterDisplay isPreview={value} CMSInput={data}/>}
    </PreviewContext.Consumer>
  )

export const FooterDisplay = ({isPreview, CMSInput}) => {
    //if isPreview is true, then the CMS user is either editing a page with the footer or editing the footer itself. 
    //if CMSInput is not an empty object, then the CMS user is editing the footer.
    let footerContent = isPreview ? (Object.keys(CMSInput).length > 0 ? CMSInput : footerPreviewContent) : getFooterContent()
    
    return (
        <footer>
            <div className="columns">
                <div className="column is-6">
                    <GetInTouch invitationText={footerContent.invitationText} />
                </div>
                <div className="column is-4 is-offset-1">
                    <ContactDetails 
                        address={footerContent.address}
                        email={footerContent.email}
                        facebookAccount={footerContent.facebookAccount}
                        twitterAccount={footerContent.twitterAccount}
                    />
                </div>
            </div>
            <div className="columns reverse-columns">
                <div className="column is-half column-align-bottom">
                    <PrivacyPolicy copyright={footerContent.copyright} />
                </div>
                <div className="column is-half">
                    <EmailSubscription 
                        emailSubscriptionText={footerContent.emailSubscriptionText}
                        emailSubscriptionLink={footerContent.emailSubscriptionLink}
                    />
                </div>
            </div>
        </footer>
    )
}

const GetInTouch = ({invitationText}) => (
    <div className="invitation-and-buttons">
        <h1 className="title is-3 is-size-4-mobile">{invitationText}</h1>
        <HeaderButtons />
    </div>
)

const ContactDetails = ({address, facebookAccount, twitterAccount}) => (
    <div className="footer-contact-us">
        <h3 className="title middle-line is-5" align="left"><span>Contact Us</span></h3>
        <div className="columns is-mobile contact-details is-vcentered">
            <div className="column address">
                <div className="content is-small address has-text-weight-semibold" align="left">
                    <p>{address.firstLine}</p>
                    <p>{address.secondLine}</p>
                    <p>{address.thirdLine}</p>
                </div>
            </div>
            <div className="column icons">
                <div className="tile is-ancestor">
                    <div className="contact tile is-parent has-text-centered">
                        <div className="tile is-4 is-child">
                            <FacebookIcon facebookAccount={facebookAccount} />
                        </div>
                        <div className="tile is-4 is-child">
                            <TwitterIcon twitterAccount={twitterAccount} />
                        </div>
                        <div className="tile is-4 is-child">
                        <ContactFormIcon/>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
)

const FacebookIcon = ({facebookAccount}) => (
    <a className="icon is-large facebook-colour"
        href={facebookAccount}
        target="_blank"
        rel="noreferrer noopener">
        <IconContext.Provider value={{size: "3em"}}>
            <FaFacebook />
        </IconContext.Provider>
    </a>
)

const TwitterIcon = ({twitterAccount}) => (
    <a className="icon is-large twitter-colour" 
        href={twitterAccount}
        target="_blank"
        rel="noreferrer noopener">
        <IconContext.Provider value={{size: "3em"}}>
            <FaTwitter />
        </IconContext.Provider>
    </a>
)

const ContactFormIcon = () => (
    <Link className="icon is-large twitter-colour has-text-primary" 
        to="/contact"
        target="_blank"
        rel="noreferrer noopener">
        <IconContext.Provider value={{size: "3em"}}>
            <FaEnvelopeOpenText />
        </IconContext.Provider>
    </Link>
)

const PrivacyPolicy = ({copyright}) => (
    <div>
        <p className="align-bottom">
            {copyright} | <Link to="/privacy"> Cookie & Privacy policies</Link>
        </p>
    </div>
)

const EmailSubscription = ({emailSubscriptionText, emailSubscriptionLink}) => (
    <a className="box has-background-primary has-text-white is-size-5"
        href={emailSubscriptionLink}
        target="_blank"
        rel="noreferrer noopener">
        {emailSubscriptionText}
    </a>
)

const getFooterContent = () => {
    const footerContent = useStaticQuery(graphql`
        query footerContent {
            markdownRemark(fields: {slug: {eq: "/header-and-footer/footer/"}}) {
                frontmatter {
                    invitationText
                    address {
                        firstLine
                        secondLine
                        thirdLine
                    }
                    email
                    facebookAccount
                    twitterAccount
                    emailSubscriptionText
                    emailSubscriptionLink
                    copyright
                }
            }
        }`
    )
    return footerContent.markdownRemark.frontmatter
}

Footer.propTypes = {
    data: PropTypes.object
}

FooterDisplay.propTypes = {
    isPreview: PropTypes.bool.isRequired,
    CMSInput: PropTypes.object
}

GetInTouch.propTypes= {
    invitationText: PropTypes.string.isRequired
}

ContactDetails.propTypes = {
    address: PropTypes.shape({ 
        firstLine: PropTypes.string,
        secondLine: PropTypes.string,
        thirdLine: PropTypes.string,
    }).isRequired,
    facebookAccount: PropTypes.string.isRequired,
    twitterAccount: PropTypes.string.isRequired,
}

FacebookIcon.propTypes = {
    facebookAccount: PropTypes.string.isRequired,
}

TwitterIcon.propTypes = {
    twitterAccount: PropTypes.string.isRequired,
}

PrivacyPolicy.propTypes = {
    copyright: PropTypes.string.isRequired,
}

EmailSubscription.propTypes = {
    emailSubscriptionText: PropTypes.string.isRequired,
    emailSubscriptionLink: PropTypes.string.isRequired,
}
