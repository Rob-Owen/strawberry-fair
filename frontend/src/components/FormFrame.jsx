import React from 'react'
import { formValidator } from './validators'
import isUrl from 'is-url'

const IFRAMEHEIGHT = 1427 //default value for the height of the form

export const FormFrame = ({form: {isPublic, link}}) => {
    if (isPublic && link) {
        // Link as url
        if (isUrl(link) && link.includes("docs.google.com/forms/")) {
            return (
                <iframe src={link} width="100%" height={IFRAMEHEIGHT} />
            )
        }
      
        // Link as iFrame
        const urlSearch = link.match(/src="(\S*)"/)
  
        if (urlSearch && urlSearch[1]) {
            const url = urlSearch[1]
  
            if (isUrl(url) && url.includes("docs.google.com/forms/")) {
                const heightSearch = link.match(/\sheight="([0-9]+)?"\s/)
                const height = (heightSearch && heightSearch[1]) ? heightSearch[1] : IFRAMEHEIGHT
  
                return (
                    <iframe src={url} width="100%" height={height}>Loadingâ€¦</iframe>
                )
            }
        }
    }

    return null
}
  
FormFrame.propTypes = { form: formValidator }