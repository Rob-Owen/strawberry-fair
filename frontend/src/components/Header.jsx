import React, {useState} from 'react'
import PreviewCompatibleImage from './PreviewCompatibleImage'


export const Header = () => (
    <header>
        <nav className="navbar is-fixed-top">
            <div className="navbar-brand">
                <a className="navbar-item" href="/">
                    <PreviewCompatibleImage imageInfo={logo} />
                </a>
                <NavBurger target="navigationBar" />
            </div>
            <NavMenu>
                {[
                    <NavDropdown title="About Us">
                        {[
                            <NavLink href="/" title="Frequently Asked Questions"/>,
                            <NavLink href="/" title="About the Fair"/>,
                            <NavLink href="/" title="Organisation"/>,
                            <NavLink href="/" title="History"/>,
                            <NavLink href="/" title="Environment"/>,
                            <NavLink href="/" title="Bucket Collection Partners"/>,
                            <NavLink href="/" title="Get Involved"/>
                        ]}
                    </NavDropdown>,
                    <NavDropdown href="/" title="Areas &amp; Events">
                        
                    </NavDropdown>,
                    <NavDropdown href="/" title="News">

                    </NavDropdown>,
                    <NavDropdown href="/" title="Traders">

                    </NavDropdown>,
                    <NavDropdown href="/" title="Support the Fair">

                    </NavDropdown>,
                    <NavDropdown href="/" title="Contact">

                    </NavDropdown>
                ]}
            </NavMenu>
        </nav>
    </header>
)


const logo = { alt: "Strawberry Fair Logo", image: "img/1-line-logo.png" }


const NavMenu = ({children}) => (
    <ul id="navigationBar" className="navbar-menu">
        <div className="navbar-start">
            {children}
        </div>
    </ul>
)


const NavBurger = ({target}) => {
    const [active, setState] = useState(false)
    const getName = () => `navbar-burger burger ${active ? "is-active" : ""}`

    return (
        <a className={getName()} data-target={target} onClick={() => setState(!active)}>
            <span></span>
            <span></span>
            <span></span>
        </a>
    )
}


const NavDropdown = ({title, children}) => {
    const [active, setState] = useState(false)
    const getName = () => `navbar-item has-dropdown ${active ? "is-active" : ""}`
    
    return (
        <li className={getName()}>
            <button className="navbar-link" onClick={() => setState(!active)}>
                {title}
            </button>
            <ul className="navbar-dropdown">
                {children}
            </ul>
        </li>
    )
}


const NavLink = ({href, title}) => (
    <li className="navbar-item">
        <a href={href}>
            {title}
        </a>
    </li>
)