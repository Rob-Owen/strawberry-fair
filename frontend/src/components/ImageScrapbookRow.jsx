import React from 'react'

import PreviewCompatibleImage from '../components/PreviewCompatibleImage'

const ImageScrapbookRow = ({scrapbookImages}) => {
  return (
    <div className="columns is-mobile">
      <RowWithTwoImages topImage={scrapbookImages.pop()} bottomImage={scrapbookImages.pop()}/>
      <div className="column is-one-third scrapbook-image">
        <PreviewCompatibleImage imageInfo={scrapbookImages.pop()} />
      </div>
      <div className="column is-one-fifth scrapbook-image">
        <PreviewCompatibleImage imageInfo={scrapbookImages.pop()} />
      </div>
      <div className="column scrapbook-image">
        <PreviewCompatibleImage imageInfo={scrapbookImages.pop()} />
      </div>
    </div>
  )
}

const RowWithTwoImages = ({topImage, bottomImage}) => (
  <div className="column is-one-fifth scrapbook-image">
    <div className="tile is-ancestor">
      <div className="tile is-vertical is-parent">
        <div className="tile is-child">
          <PreviewCompatibleImage imageInfo={topImage} />
        </div>
        <div className="tile is-child">
          <PreviewCompatibleImage imageInfo={bottomImage} />
        </div>
      </div>
    </div>
  </div>
)


export default ImageScrapbookRow
