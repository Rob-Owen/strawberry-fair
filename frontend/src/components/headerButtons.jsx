import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'

import { InternalLink } from './InternalLink'
import { PreviewContext } from '../util/context.jsx'
import headerButtonPreviewData from '../data/headerButtonPreviewData'
import PropTypes from 'prop-types'

const HeaderButtons = () => (
  <PreviewContext.Consumer>
    {value => <HeaderButtonsDisplay isPreview={value} />}
  </PreviewContext.Consumer>
)

export default HeaderButtons

const HeaderButtonsDisplay = ({isPreview}) => {
  const data = isPreview ? headerButtonPreviewData : getHeaderButtonLinksAndText()

  return (
    <div className="black-and-red buttons ">
        <InternalLink to={data.blackButton.frontmatter.link} className="button is-secondary">{data.blackButton.frontmatter.text}</InternalLink>
        <InternalLink to={data.redButton.frontmatter.link} className="button is-primary">{data.redButton.frontmatter.text}</InternalLink>
    </div>
  )
}

function getHeaderButtonLinksAndText() {
  return useStaticQuery(graphql`
    query headerButtons {
      redButton: markdownRemark(fileAbsolutePath: { regex: ".*/src/pages/header-and-footer/buttons/red-button.md/"}) {
        frontmatter {
          link
          text
        }
      }
      blackButton: markdownRemark(fileAbsolutePath: { regex: ".*/src/pages/header-and-footer/buttons/black-button.md/"}) {
          frontmatter {
            link
            text
          }
        }
    }`
  )
}

HeaderButtonsDisplay.propTypes = {
  isPreview: PropTypes.bool.isRequired,
}
