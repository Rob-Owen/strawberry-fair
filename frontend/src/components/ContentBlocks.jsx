import React from 'react'
import ImageScrapbookRow from './ImageScrapbookRow'

const ContentBlocks = ({contentBlocks, contentBlocksHtml, BodyComponent}) => (
  <React.Fragment>
    {contentBlocks && contentBlocks.map((block, index) => (
        <ContentBlock 
          contentTitle={block.contentTitle}
          scrapbookImages={block.scrapbookImages}
          content={contentBlocksHtml[index]}
          sideSnippet={block.sideSnippet}
          BodyComponent={BodyComponent}
          key={index}
        />
      )
    )}
    </React.Fragment>
  )

export default ContentBlocks

export const ContentBlock = ({contentTitle, scrapbookImages, content, sideSnippet, BodyComponent}) => (
<section className="section">
  <ImageScrapbookRow scrapbookImages={scrapbookImages}/>
  <h1 className="title">{contentTitle}</h1>
  <BodyComponent content={content}/>
</section>
)
