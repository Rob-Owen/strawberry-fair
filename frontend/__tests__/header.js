import React from "react"


import { Header } from "../src/components/Header"
import { render, fireEvent } from "@testing-library/react"

test("Dropdown menus closed on loading", () => {
  const { getByText } = render(<Header />)

  expect(getByText("Frequently Asked Questions")).not.toBeVisible()
})

test("Dropdown menus can open", async () => {
  const { getByText } = render(<Header />)

  fireEvent.click(getByText("About Us"))

  await expect(getByText("Frequently Asked Questions")).toBeVisible()
})