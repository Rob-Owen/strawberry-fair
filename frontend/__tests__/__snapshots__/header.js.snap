// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <nav
    className="navbar is-fixed-top"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="/"
      >
        <img
          alt="Strawberry Fair Logo"
          src="/img/1-line-logo.png"
          style={
            Object {
              "borderRadius": "5px",
            }
          }
        />
      </a>
      <a
        className="navbar-burger burger "
        data-target="navigationBar"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </a>
    </div>
    <ul
      className="navbar-menu "
      id="navigationBar"
    >
      <div
        className="navbar-start"
      >
        <li
          className="navbar-item has-dropdown "
        >
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <div>
                <button
                  className="button"
                  onClick={[Function]}
                >
                  About Us
                </button>
              </div>
            </div>
            <ul
              className="navbar-dropdown is-hidden-touch"
            >
              <li
                className="navbar-item"
              >
                <a
                  className="dropdown-item"
                  href="/"
                >
                  Frequently Asked Questions
                </a>
              </li>
              <li
                className="navbar-item"
              >
                <a
                  className="dropdown-item"
                  href="/"
                >
                  About the Fair
                </a>
              </li>
              <li
                className="navbar-item"
              >
                <a
                  className="dropdown-item"
                  href="/"
                >
                  Organisation
                </a>
              </li>
              <li
                className="navbar-item"
              >
                <a
                  className="dropdown-item"
                  href="/"
                >
                  History
                </a>
              </li>
              <li
                className="navbar-item"
              >
                <a
                  className="dropdown-item"
                  href="/"
                >
                  Environment
                </a>
              </li>
              <li
                className="navbar-item"
              >
                <a
                  className="dropdown-item"
                  href="/"
                >
                  Bucket Collection Partners
                </a>
              </li>
              <li
                className="navbar-item"
              >
                <a
                  className="dropdown-item"
                  href="/"
                >
                  Get Involved
                </a>
              </li>
            </ul>
          </div>
        </li>
        <li
          className="navbar-item has-dropdown "
        >
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <div>
                <button
                  className="button"
                  onClick={[Function]}
                >
                  Areas & Events
                </button>
              </div>
            </div>
            <ul
              className="navbar-dropdown is-hidden-touch"
            >
              <li
                className="navbar-item"
              >
                <a
                  className="dropdown-item"
                  href="/calendar"
                >
                  Calendar
                </a>
              </li>
            </ul>
          </div>
        </li>
        <li
          className="navbar-item has-dropdown "
        >
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <div>
                <button
                  className="button"
                  onClick={[Function]}
                >
                  News
                </button>
              </div>
            </div>
            <ul
              className="navbar-dropdown is-hidden-touch"
            />
          </div>
        </li>
        <li
          className="navbar-item has-dropdown "
        >
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <div>
                <button
                  className="button"
                  onClick={[Function]}
                >
                  Traders
                </button>
              </div>
            </div>
            <ul
              className="navbar-dropdown is-hidden-touch"
            />
          </div>
        </li>
        <li
          className="navbar-item has-dropdown "
        >
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <div>
                <button
                  className="button"
                  onClick={[Function]}
                >
                  Support the Fair
                </button>
              </div>
            </div>
            <ul
              className="navbar-dropdown is-hidden-touch"
            />
          </div>
        </li>
        <li
          className="navbar-item has-dropdown "
        >
          <div
            className="dropdown"
          >
            <div
              className="dropdown-trigger"
            >
              <div>
                <button
                  className="button"
                  onClick={[Function]}
                >
                  Contact
                </button>
              </div>
            </div>
            <ul
              className="navbar-dropdown is-hidden-touch"
            />
          </div>
        </li>
      </div>
    </ul>
  </nav>
</header>
`;
